src/createServer.ts

```diff
@@ -106,6 +107,11 @@ export default function (
           res.end(JSON.stringify('OK'))
           break
         case '/upload':
+          // Do not return an error status code, just ignore the upload request entirely
+          if (opts.ignoreUploadRequests) {
+            res.end(JSON.stringify('IGNORED'))
```

Comment:
I don't like this approach. This is a silent error. If you think pnpm should not fail in this case, just put the upload method into a try/catch in supi and log a warning.